### PR TITLE
Records total patch

### PR DESCRIPTION
**Pull request by nbey**
Fix for returning the correct total of records.

### CODE REVIEW

**Comment by themightychris** [comment]
@nbey what problem were you solving here? The total returned from the API is supposed to be the count of all results across all pages not just the current page. data.length tells you how many records you have in the current response it doesn't need to be mirrored in an integer next to it

**Comment by nbey** [comment]
I get this response when from those methods when the table does not exist.

>{
success: true,
data: [ ],
conditions: [ ],
total: 1,
limit: false,
offset: false
}


>The totals returned from the methods: handleBrowseRequest, handleQueryRequest
are calculated by DB::foundRows(), which falsely returns: 1, after attempting to querying the DB for a table that doesn't exist.
This patch simply checks if there are any records returned, before attempting to calculate with DB::foundRows();

**Comment by jmealo** [comment]
The error counts as a row being returned. Should we be handling errors differently? I don't expect success to be ``true`` either.

**Comment by themightychris** [comment]
`success` should be true because with ActiveRecord it is normal for a table to not exist until the first record is written. Therefor a nonexistent table isn't a failure, it's just an empty result set. As far as the client is concerned it got what it requested successfully.

**Comment by themightychris** [comment]
I wonder why `SELECT FOUND_ROWS()` returns `1` after a SELECT against a nonexistent table, is it because that's just the correct result for the last successful query before the one that just failed, or does it actually get set to `1` when a query fails.

I'm thinking it would be best if our implementation of [`DB::foundRows()`](https://github.com/JarvusInnovations/Emergence/blob/master/php-bootstrap/lib/DB.class.php#L37) skipped making the FOUND_ROWS query and just returned 0 if the last query failed. -- @jmealo any thoughts?

**Comment by nbey** [comment]
>Beyond the cases described here, the behavior of FOUND_ROWS() is undefined (for example, its value following a SELECT statement that fails with an error).

http://dev.mysql.com/doc/refman/5.0/en/information-functions.html#function_found-rows

**Comment by themightychris** [comment]
@nbey could you take a stab and a patch for foundRows over in the [Emergence](https://github.com/JarvusInnovations/Emergence) repo?

**Comment by jmealo** [comment]
``FOUND_ROWS`` doesn't work with replication either.

File: php-classes/RecordsRequestHandler.class.php [hunk 1]
```diff
@@ -201,27 +201,29 @@ static public function handleQueryRequest($query, $conditions = array(), $option
                 $options['order'] = $className::mapFieldOrder(static::$browseOrder);
         }
 
+        $responseData = $className::getAllByQuery(
+			'SELECT DISTINCT %s %s FROM `%s` %s %s WHERE (%s) %s %s %s'
+			,array(
```
Comment (jmealo) [inline php-classes/RecordsRequestHandler.class.php]: new style array?

**Comment by nbey** [comment]
I may have found a solution, i'll test it on one of my servers and get back.

**Comment by jmealo** [comment]
We should still handle the table not found exception separately from other
errors. I was under the impression when an error is returned that it counts
as a row. You can see this behavior in MySQL workbench. All other errors
should return success: false.

**Comment by themightychris** [comment]
We should and I believe we already do, but DB::foundRows returns information about the previous query so it's not directly effected by any exception thrown in the previous query. Given that mysql's docs say that FOUND_ROWS() behavior is undefined if the previous query failed, I think its reasonable for it to return 0 if the previous query failed without calling FOUND_ROWS()

**Comment by nbey** [comment]
I committed a patch on my fork that I tested and worked -- my solution was to store the last query log in memory, and return `0` when MYSQL returns `-1` for `affected_rows`, and we have an error logged.

You can see it here: https://github.com/nbey/Emergence/commit/728c5e354924dbc04370f3755f154ba65e35e7bd

**Comment by themightychris** [comment]
@nbey I think we could just use http://php.net/manual/en/mysqli.errno.php or http://php.net/manual/en/mysqli.sqlstate.php instead of storing anything ourselves

**Comment by jtwiest** [comment]
Made a new PR for this ticket that took the sqlstate approach.
https://github.com/JarvusInnovations/Emergence/pull/32

**Comment by themightychris** [comment]
Fixed by JarvusInnovations/Emergence#32

**Pull request closed without merging**